spaghetti/util.py

```diff
@@ -762,20 +761,16 @@ def _arcs_as_gdf(net, points, id_col=None, geom_col=None):
 
     """
 
-    # arcs
-    arcs = {}
-
-    # iterate over network arcs
-    for vtx1_id, vtx2_id in net.arcs:
-        # extract vertices comprising the network arc
-        vtx1 = points.loc[(points[id_col] == vtx1_id), geom_col].squeeze()
-        vtx2 = points.loc[(points[id_col] == vtx2_id), geom_col].squeeze()
-        # create a LineString for the network arc
-        arcs[(vtx1_id, vtx2_id)] = LineString((vtx1, vtx2))
+    def _line_coords(loc):
+        return (
+            (points.loc[loc[0]].geometry.x, points.loc[loc[0]].geometry.y),
+            (points.loc[loc[1]].geometry.x, points.loc[loc[1]].geometry.y),
+        )
 
     # instantiate GeoDataFrame
-    arcs = geopandas.GeoDataFrame(
-        sorted(list(arcs.items())), columns=[id_col, geom_col]
+    arcs = pandas.DataFrame(zip(sorted(net.arcs)), columns=[id_col])
+    arcs = arcs.set_geometry(
+        shapely.linestrings(arcs[id_col].map(_line_coords).values.tolist())
```

Comment:
I suppose there should be a way to vectorise this via numpy magic. Would need to look into it if you want a suggestion.

Reply:
_Very_ welcome to suggestions. I had thought that use `shapely.linestrings()` was magic in itself. 😄

Reply:
Let's keep it, it may be unnecessarily complex and unreadable.